packages/backend/src/core/sync/SyncScheduler.ts

```diff
@@ -26,7 +27,10 @@ export class SyncScheduler {
   constructor(
     private readonly syncStatusRepository: SyncStatusRepository,
     private readonly blockDownloader: BlockDownloader,
-    private readonly dataSyncService: DataSyncService | ValidiumDataSyncService,
+    private readonly dataSyncService:
```

Comment:
This type should probably be given as an interface that all sync services implement